tests/test_hf_api.py

```diff
@@ -4239,3 +4231,20 @@ def test_upload_large_folder(self, repo_url: RepoUrl) -> None:
             for j in range(N_FILES_PER_FOLDER):
                 assert f"subfolder_{i}/file_lfs_{i}_{j}.bin" in uploaded_files
                 assert f"subfolder_{i}/file_regular_{i}_{j}.txt" in uploaded_files
+
+
+class TestHfApiAuthCheck(HfApiCommonTest):
+    @use_tmp_repo(repo_type="dataset")
+    def test_auth_check_success(self, repo_url: RepoUrl) -> None:
+        self._api.auth_check(repo_id=repo_url.repo_id, repo_type=repo_url.repo_type)
+
+    def test_auth_check_repo_missing(self) -> None:
+        with self.assertRaises(RepositoryNotFoundError):
+            self._api.auth_check(repo_id="username/missing_repo_id")
+
+    def test_auth_check_gated_repo(self) -> None:
+        gated_repo_id = "gated_repo_id"
+        self._api.create_repo(repo_id=gated_repo_id, token=TOKEN, gated=True).repo_id
```

Comment:
Setting a repo as "gated" is not possible to do directly in `create_repo`. You can check here how to do it: https://github.com/huggingface/huggingface_hub/blob/ecbbeb355aadcbe683c604e3bab27ac4a6e33d4c/tests/test_hf_api.py#L3991